Sources/TurboConfig.swift

```diff
@@ -11,12 +11,14 @@ public class TurboConfig {
 
     /// Optionally customize the web views used by each Turbo Session.
     /// Ensure you return a new instance each time.
-    public var makeCustomWebView: WebViewBlock?
+    public var makeCustomWebView: WebViewBlock = { webView in
+        return webView
+    }
 
     // MARK: - Internal
 
     func makeWebView() -> WKWebView {
-        makeCustomWebView?() ?? WKWebView(frame: .zero, configuration: makeWebViewConfiguration())
+        makeCustomWebView(WKWebView(frame: .zero, configuration: makeWebViewConfiguration()))
```

Comment:
In our application's case, it felt like we'd get the most value out of re-using the configuration instance built by the `makeWebViewConfiguration()` call (since it as the User Agent already set, shares a Process Pool, etc).

Maybe it'd be better if the closure were changed to provide the configuration instead:

```swift
public typealias WebViewBlock = (_ configuration: WKWebViewConfiguration) -> WKWebView

public var makeCustomWebView: WebViewBlock = { configuration in
      return WKWebView(frame: .zero, configuration: configuration)
}

func makeWebView() -> WKWebView {
      makeCustomWebView(makeWebViewConfiguration()))
}
```

Reply:
I've also opened https://github.com/joemasilotti/TurboNavigator/pull/42. That could either work hand-in-hand with this change as a complement, or be an alternative.

Reply:
I think passing the configuration and requiring the developer to build the web view is the way to go. That provides the most flexibility _and_ default configuration (user agent and process pool).

Rereading the [`WKWebViewConfiguration` docs](https://developer.apple.com/documentation/webkit/wkwebviewconfiguration) again, it looks like you _must_ assign it when the web view is instantiated for anything to apply.

> You create a WKWebViewConfiguration object in your code, configure its properties, and pass it to the initializer of your [WKWebView](https://developer.apple.com/documentation/webkit/wkwebview) object. **The web view incorporates your configuration settings only at creation time; you cannot change those settings dynamically later.**

Reply:
Thanks you for that feedback and guidance!

> The web view incorporates your configuration settings only at creation time; you cannot change those settings dynamically later.

In my experience, that isn't true for variables that are accessible through the configuration itself, like the `userContentController` that the [turbo-ios](https://github.com/hotwired/turbo-ios/blob/7.0.0/Source/WebView/WebViewBridge.swift#L41-L49) package utilizes to inject its script:

```swift
    init(webView: WKWebView) {
        self.webView = webView
        setup()
    }
    
    private func setup() {
        webView.configuration.userContentController.addUserScript(userScript)
        webView.configuration.userContentController.add(ScriptMessageHandler(delegate: self), name: messageHandlerName)
    }
```

Those assignments occur _after_ the WebView has been constructed.

Reply:
That's correct. The objects that the configuration holds on to can be assigned at will. But the _properties_ of the configuration cannot.

You can test this by trying to override `applicationNameForUserAgent`.

```swift
let configuration = WKWebViewConfiguration()
configuration.applicationNameForUserAgent = "BEFORE"

let webView = WKWebView(frame: .zero, configuration: configuration)
webView.configuration.applicationNameForUserAgent = "AFTER"

print(webView.configuration.applicationNameForUserAgent) // Optional("BEFORE")
```